### DISCUSSION TITLE
Add function to Resource filterable{Model} as we have relatable{Model} and respect Policy

### DISCUSSION THREAD

**Discussion by sirikkoster** [laravel/nova-issues#4771] [Ideas & Feature Requests]
Is it possible to get a function so we can filter the "->filterable()" function and respect the Policies.

Sometimes i got options which are not available for a specific Resource, and doesn't want to show in the select

**Comment by juse-less** [comment]
The `relatable*()` methods are already meant for the `Field::filterable()` method (and select Fields in forms), AFAIK.
The initial section of [the documentation](https://nova.laravel.com/docs/4.0/resources/authorization.html#relatable-filtering) states that
> If you would like to filter the queries that are used to populate relationship model selection menus, you may override the relatableQuery method on your resource.

Which would also then be applied to the dynamic methods [shown right below](https://nova.laravel.com/docs/4.0/resources/authorization.html#dynamic-relatable-methods).

I'm not sure if they use the Policies at this time, though, but it is listed in the Authorization section of the documentation.

I spent a few hours yesterday, trying to filter the `HasMany`, and `HasOne::ofMany()` panels using those methods before I came to the conclusion, since those queries weren't reflected there.
I.e., the result shown in the `HasMany` panel/table were unfiltered.

I'd too like to filter relationship fields selection menus, so this came as a happy surprise.
Just gotta get the time to fix that. 🙂
